### DISCUSSION TITLE
Backing out GHI from Solar Power?

### DISCUSSION THREAD

**Discussion by Stratix11** [pvlib/pvlib-python#1305] [Q&A] [answered]
Hey everyone! Long-time lurker and user of pvlib, and it's been a great tool in my research! 

I wanted to ask if anyone here has had experience converting solar power to resource (GHI)? Is this possible with pvlib, or even possible at all? My initial searches and discussions with colleagues hasn't led me anywhere, so I thought I would ask the general pvlib community. This is assuming I know everything I need to know about the solar panels and array. Thanks!

**Comment by kandersolar** [comment, marked as answer]
Two ideas, neither of which I'm confident would work well:

1. Construct a ModelChain that describes your system and use a numerical optimizer (e.g. scipy.optimize) to find the GHI that best predicts the measured power.  
2. Invert each step in the modeling process (analytically where possible, numerically otherwise) and run the traditional modeling process in reverse.  This seems like a lot of work to me.  There is at least [gti_dirint](https://pvlib-python.readthedocs.io/en/stable/generated/pvlib.irradiance.gti_dirint.html) for estimating GHI from POA.  I guess you could invert a simple model like PVWatts without much headache.

In either case I suspect the uncertainty in the estimated GHI would be large enough that it would be a lot easier to just use GHI from a dataset like PSM3 or ERA5 or something.  What's the use case for estimating irradiance from measured power?

**Comment by williamhobbs** [comment]
I'm curious if this concept went anywhere. 

Could you model clear sky power for the PV system, divide that into actual power to get a clear sky power index, and then use that to estimate GHI? You would have to assume clear sky GHI index is the same as the index for power (or come up with a better translation). And if the PV systems spends time clipping (i.e., at the AC inverter nameplate) then you can only estimate the minimum GHI during those periods.

**Reply by wholmgren** [reply]
power / clearsky_power is more closely related to poa / clearsky_poa. Side note: I actually implemented `gti_dirint` so that I could back out power to GHI for use with `detect_clearsky`. I just assumed constant inverter efficiency and then inverted (ha) the pvwatts DC model to get POA. It all worked ok for my purposes.

> And if the PV systems spends time clipping (i.e., at the AC inverter nameplate) then you can only estimate the minimum GHI during those periods.

Correct. Doesn't work during those times, so I used a simple filter for the clipped periods (probably would now look at pvanalytics for that filter).

**Reply by williamhobbs** [reply]
Yeah, you would have to assume poa / clearsky_poa is close enough to ghi / clearsky_ghi.
